### DISCUSSION TITLE
Error details: The 'Microsoft.Ace.OLEDB.12.0' provider is not registered on the local machine.

### DISCUSSION THREAD

**Discussion by Madhavivsp** [actions/virtual-environments#5731] [Installed software]
We are getting this  error for our validations run pipelines from couple of days.  
"The unit test adapter failed to connect to the data source or to read the data. For more information on troubleshooting this error, see "Troubleshooting Data-Driven Unit Tests" . Error details: The 'Microsoft.Ace.OLEDB.12.0' provider is not registered on the local machine."

Our 1ES support team mentioned that it is related to images. We have not changed anything recently in code or test data. This error started triggering from 3 days. Any updated happened to images? we are blocked to proceed our test case validations. Please help us to unblock

we are using MMS2016 image

**Comment by al-cheb** [comment]
Hey @Madhavivsp,
We don't support Windows Server 2016 anymore - https://github.com/actions/virtual-environments/issues/5403

**Reply by Madhavivsp** [reply]
Hi @al-cheb , do we need to update our image to windows higher version to get rid of that error?

**Comment by al-cheb** [comment]
@Madhavivsp, if you can reproduce the issue on Windows Server 2019/2022  Hosted Agents/GitHub feel free to create a new issue in our repo. We will investigate it from our side if you provide repro steps with a sample repo.
